src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -348,6 +349,31 @@ public AkkaConfigurationBuilder WithHealthCheck(AkkaHealthCheckRegistration regi
             return this;
         }
         
+        /// <summary>
+        /// Registers a DI-resolved health check with the <see cref="AkkaConfigurationBuilder"/>.
+        /// The health check type must be registered in the DI container.
+        /// </summary>
+        /// <param name="name">The healthcheck name.</param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported upon failure of the health check. If the provided value
+        /// is <c>null</c>, then <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used for filtering health checks.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <typeparam name="T">The type of the health check that implements <see cref="IAkkaHealthCheck"/>.</typeparam>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        public AkkaConfigurationBuilder WithHealthCheck<T>(string name, HealthStatus? failureStatus = null, 
+            IEnumerable<string>? tags = null, TimeSpan? timeout = null) where T : class, IAkkaHealthCheck
+        {
+            // Create a health check instance that will be resolved from DI when needed
+            var registration = new AkkaHealthCheckRegistration(name, GetServiceOrCreateInstance, failureStatus, tags, timeout);
+            
+            HealthChecks[registration.Name] = registration;
+            return this;
+            
+            static T GetServiceOrCreateInstance(IServiceProvider sp) => ActivatorUtilities.GetServiceOrCreateInstance<T>(sp);
```

Comment:
This is the big time-saver - it uses the `ActivatorUtilities` from MSFT.EXT.DI to handle the `IServiceProvider` resolution / instantiation of CTOR arguments for us, so we're not having to write that code ourselves. This is the same thing that powers the DI'd `Props` in Akka.DependencyInjection.